koku/api/common/pagination.py

```diff
@@ -306,20 +306,32 @@ def get_count(self, queryset):
         return len(queryset.get("data", [])[0].get("resource_ids", []))
 
     def get_paginated_data(self, queryset):
-        """Special case pagination for EC2 resource IDs."""
+        """
+        Pagination EC2 resource IDs based on the request type.
+
+        Args:
+        queryset (dict): The data containing resource IDs to be paginated.
+
+        Returns:
+        paginated_data (list): If the request expects CSV data, returns a slice of resource IDs.
+                                Otherwise, list containing a single dictionary with paginated resource IDs.
+        """
 
         paginated_data = []
 
-        for item in queryset.get("data", []):
-            resource_ids = item.get("resource_ids", [])
-            resource_count = len(resource_ids)
+        data = queryset.get("data", [])[0]  # only single month data expected
```

Comment:
If `queryset.get("data", [])` returns an empty list accessing 0 will raise an IndexError.

Reply:
Yes thanks for catching that, updated it